libgalois/include/katana/PropertyFileGraph.h

```diff
@@ -109,6 +114,42 @@ class KATANA_EXPORT PropertyFileGraph {
   // caller of SetTopology.
   GraphTopology topology_;
 
+  // Keep partition_metadata, master_nodes, mirror_nodes out of the public interface,
+  // while allowing Distribution to read/write it for RDG
+  friend class Distribution;
+  const tsuba::PartitionMetadata& partition_metadata() const {
+    return rdg_.part_metadata();
+  }
+  void set_partition_metadata(const tsuba::PartitionMetadata& meta) {
+    rdg_.set_part_metadata(meta);
+  }
+
+  /// Per-host vector of master nodes
+  ///
+  /// master_nodes()[this_host].empty() is true
+  /// master_nodes()[host_i][x] contains LocalNodeID for my master and
+  ///   host_i has a mirror
+  const std::vector<std::shared_ptr<arrow::ChunkedArray>>& master_nodes()
+      const {
+    return rdg_.master_nodes();
+  }
+  void set_master_nodes(std::vector<std::shared_ptr<arrow::ChunkedArray>>&& a) {
+    rdg_.set_master_nodes(std::move(a));
+  }
+
+  /// Per-host vector of mirror nodes
+  ///
+  /// mirror_nodes()[this_host].empty() is true
+  /// mirror_nodes()[host_i][x] contains LocalNodeID for my mirrors and
+  ///   host_i has the master
```

Comment:
same nit: "mirrors that have the master on host_i" or "mirrors for which host_i has the master"

Reply:
And again.